### PR TITLE
Add Logging options to IPersistenceConfiguration.

### PR DESCRIPTION
**Pull request by fredrikhr**
### `@azure/msal-node-extensions`

Adds support for configuring `LoggerOptions` when creating an `IPersistence` using `PersistenceCreator.createPersistence`.

## Description

All types implementing `IPersistence` in the `@azure/msal-node-extensions` package support a `LoggerOptions` object as an optional parameter to their static `create` methods. `PersistenceCreator.createPersistence`, the factory method creating instances of these types, has currently no way of passing such a `LoggerOptions` object in to these `create` methods.

In cases when a custom logger callback is used in MSAL, it is desirable that logging is done consistently across all components being used in MSAL. This PR adds a way for users to customize the logger, allowing for a consistent logging experience.

## Implementation

Add a new optional property `loggerOptions` to the `IPersistenceConfiguration` interface type. Inside `PersistenceCreator.createPersistence` this property is simply passed along to all `create` methods that are called within the factory.

## Breaking changes

None. The added property in `IPersistenceConfiguration` is the only noticeable change to the API surface. If that property is left `undefined` the previous behavior still applies.

### CODE REVIEW

**Comment by tnorling** [comment]
@fredrikhr Thanks for the PR! Can you please run `npm run beachball:change` from the root of the repo and commit the changefile it generates? Thanks!

**Comment by fredrikhr** [comment]
@tnorling Does [4f4cf9e](https://github.com/AzureAD/microsoft-authentication-library-for-js/pull/5626/commits/4f4cf9e25efa2c2f34eb3e3c9f702d050039a4bd) look fine?

File: change/@azure-msal-node-extensions-8668eb8b-ec9f-46cd-a712-5c59e9cf5f5e.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+  "type": "prerelease",
+  "comment": "Add Logging options to IPersistenceConfiguration.",
```
Comment (derisen) [inline change/@azure-msal-node-extensions-8668eb8b-ec9f-46cd-a712-5c59e9cf5f5e.json]:
```suggestion
  "comment": "Add Logging options to IPersistenceConfiguration #5626",
```

**Review by tnorling — APPROVED** [review]

**Review by derisen — APPROVED** [review]
LGTM, thanks @fredrikhr -one suggestion

**Pull request merged**